### PR TITLE
Improve Code Quality

### PR DESCRIPTION
**Pull request by iamcarbon**
This PR includes a handful of non-breaking code quality improvements, nits.

- Makes a few readonly structs, readonly, and passes a larger readonly struct by readonly ref
- Updates the target lang to 11
- Replaces the NET45 symbol with NETFRAMEWORK
- Removes a redundant { } scope
- Marks a few private helper methods as static
- Eliminates a few allocations in ZipArchiveFileSystem

### CODE REVIEW

File: src/Zio/IFileSystem.cs [hunk 1]
```diff
@@ -2,235 +2,232 @@
 // This file is licensed under the BSD-Clause 2 license. 
 // See the license.txt file in the project root for more information.
 
-using System;
```
Comment (xoofx) [inline src/Zio/IFileSystem.cs:5]: Weird big diff in this file, maybe EOL?

Comment (iamcarbon) [reply src/Zio/IFileSystem.cs:5]: I had missed this file when updating the library to file-scoped namespaces in a previous PR. The line changes are related to enabling FSN's and reducing the indentation level.

**Comment by iamcarbon** [comment]
@xoofx Ready for review.

**Comment by xoofx** [comment]
Thanks!

**Pull request merged**